### DISCUSSION TITLE
Connecting to the server behind a stormShield firewall

### DISCUSSION THREAD

**Discussion by AmiardBrice** [rustdesk/rustdesk-server-pro#424] [General]
Hello,

We have just installed a Pro version of RustDesk on our server.
In most cases we have no problems.

However, for some of our customers where we want to install 'the host version' the connection is impossible.
The particularity of these customers: their network infrastructure includes a stormShield firewall upstream (https://www.stormshield.com/ ). It is a kind of "advanced" firewall that does a lot of things and among other things DPI. And obviously we do not have access to the interface of these stormShield

The error message is relatively clear. "network XXXX inaccessible".
Similarly we cannot connect the application to the user account (socket timeout).

Have you ever encountered this kind of problem?

Note: TeamViwer Host works behind a stormShield

Thanks in advance

**Comment by rustdesk** [comment]
We have never encountered this.

**Reply by rustdesk** [reply]
You can try `http` proxy, https://github.com/rustdesk/rustdesk/wiki/Set-up-http-proxy-server, it may help.
Some users use `http` proxy in restricted network. https://github.com/rustdesk/rustdesk-server-pro/discussions/300

**Reply by AmiardBrice** [reply]
Thanks, it works (with a proxy listening on port 443) !.

All other ports are probably closed externally by default in the Stormshield firewall

**Comment by AmiardBrice** [comment]
Hi,

We have again somme acces error on some server behind a "complex firwall" (such as stormShield, cisco, or cosy), event with a 443 proxy firwall.

I would like to know if rustdesk use some other port (21115, 21118, ...) event with the 443 proxy.
If not, could this problem be due to some services like DPI (paquet inspection event with https ???).

PS : on most of our clients (event on some client with a stormShield firewall),  the proxy on 443 solved our access problems

**Reply by rustdesk** [reply]
> I would like to know if rustdesk use some other port (21115, 21118, ...) event with the 443 proxy.

All port will go 443 proxy.

**Reply by briceAmiard** [reply]
I still have some limitation, it may be due tu an inspection mayde by the firwall.

I'm using proxy on 443 (allawy's open on output) with HTTP, And some firwall inspect trame content  and port matching, witch is a probleme in out case (non SSL on 443).

Probleme we already have ~80 customer setteed with a proxy http://WAN_IP:443 .

I want  try a config with HAProxy (wich can redirect trafic on some condition : 443 with SSL/TLS => Stunnel => rustdesk server , 443 without SSL/TLS =>  rustDesk).
For doing this test i've try to install rustDesk with our licence key (and HAProxy, stunnel,....) on an other VM.
And of course our licence is already in use..., could it be possible to have a short term licence (1 or 2 weeks) for doing this test (by mail) ? 

Thanks in advance

**Comment by AmiardBrice** [comment]
j'ai créé deux version d'executable, l'une en http (utilisé par defaut), et l'autre en https. Ainsi j'économise un peut de ressource CPU, en ne passant pas la phase encodage/décodage que lorsque c'est nécessaire.

Et je n'ai pas contacter de bug avec le HTTPS (contrairement à ce que l'on peut lire sur d'autre post avec la methode HTTPS via apache/gninx)
